### DISCUSSION TITLE
Reading objects in Hfss3DLayout (pyaedt)

### DISCUSSION THREAD

**Discussion by neeraj-lohia** [ansys/pyaedt#5553] [Q&A] [answered]
I am working on pyaedt (python based ansys modelling) and I have a Hfss3DLayout project. I want to read the objects on a particular layer.

```
import ansys.aedt.core as pyaedt
project = pyaedt.Hfss3dLayout(r"C:\Scripts\work\my_project.aedt")
object_layer1 = project.modeler.objects_by_layer("Layer_1")

```
With the above function, I can get the list of object names but not the actual objects.

In Q3D and HFSS, there is a function project.modeler.get_object_from_name("objectName"), which can be used to get objects using the name of the object, but this function does not exist for Hfss3DLayout.

How can I read the object in Hfss3DLayout?

**Comment by Samuelopez-ansys** [comment, marked as answer]
@neeraj-lohia You can get the object just doing:

project.modeler.geometries[object_layer1[0]]

I encourage you to debug with an IDE like Pycharm or Visual Studio, it will help you to identify which properties and method can help you.

Samuel

**Reply by neeraj-lohia** [reply]
"Thank you for the suggestion, Samuel. However, I've found that using project.modeler.geometries[object_layer1[0]] only returns objects with predefined geometries like rectangles or circles. It doesn't seem to work for objects with strange shapes, such as bondwires or user-defined shapes.
